drivers/sip_svc/sip_svc_intel_socfpga.c

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2022-2023, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Intel SoC FPGA platform specific functions used by ARM SiP Services.
+ */
+
+#include <string.h>
+#include <zephyr/drivers/sip_svc/sip_svc_agilex_mailbox.h>
+#include <zephyr/drivers/sip_svc/sip_svc_agilex_smc.h>
+#include <zephyr/drivers/sip_svc/sip_svc_driver.h>
```

Comment:
the `sip_svc_driver.h` header should go into `include/zephyr/drivers`.

Reply:
also it should be called `sip_svc.h`

Reply:
There is already a sip_svc.h file for the subsystem public API's. The users are only required to use the public subsystem API's. The sip_svc_driver.h contains the API format for platform specific implementation for sip_svc . So I can move it outside of include/zephyr/drivers/sip_svc/, But this can be renamed to **sip_svc_plat.h**?Is this fine?